public/js/reducers/sources.js

```diff
@@ -100,10 +96,19 @@ function update(state = State(), action: Action) : Record<SourcesState> {
       }
 
       return _updateText(state, action, [action.originalSource]);
+
     case "NAVIGATE":
-      // Reset the entire state to just the initial state, a blank state
-      // if you will.
-      return State();
+      if (!action.data) {
+        return state;
+      }
+      if (!state.url) {
+        return State()
+          .set("url", action.data.url);
+      }
+      if (state.url === action.data.url) {
+        return state;
+      }
+      return State().set("url", action.data.url);
```

Comment:
@jasonLaster do you have any suggestion for refreshing the target debuggee in cypress without refreshing the debugger (for the integration tests)?